tests/Foliage/Tests/Utils.hs

```diff
@@ -0,0 +1,47 @@
+module Foliage.Tests.Utils (
+  checkRequiredProgram,
+  callCommand,
+  readCommand,
+  withFixture,
+) where
+
+import Control.Exception (bracket)
+import Control.Monad (when)
+import Data.Foldable (for_)
+import Data.Functor (void)
+import Data.Maybe (isNothing)
+import System.Directory
+import System.FilePath
+import System.Posix.Temp (mkdtemp)
+import System.Process (readCreateProcess, shell)
+
+-- | Set up a temporary directory prepopulated with the fixture files
+withFixture :: FilePath -> IO () -> IO ()
+withFixture name =
+  bracket acquire release . const
+ where
+  acquire = do
+    cur <- getCurrentDirectory
+    fixtureDir <- makeAbsolute name
+    workDir <- mkdtemp $ fixtureDir ++ "."
+    setCurrentDirectory workDir
```

Comment:
This was surprising to me. Can we rather just return the workdir from `bracket`, and then make it the responsibility of `action` to cd there and back?

Reply:
I guess it sort of makes sense since you can "clean up" in release... :thinking:

Reply:
Should be in the haddock at least

Reply:
I went through a bunch of different ways to do it and in the end I landed on this :shrug:

Reply:
I just tried your first suggestion and it didn't work: if the test changes directory it has to remember to change directory back to where it was otherwise the following test will find itself lost (tests/fixtures/simple is a relative path). I'll rename the function as you suggested.

Reply:
Notice it has to make clear that it's not the fixture directory but a temporary work directory.